include/exiv2/bmffimage.hpp

```diff
@@ -0,0 +1,177 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, 5th Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+// *****************************************************************************
+#include "exiv2lib_export.h"
+
+// included header files
+#include "image.hpp"
+
+// *****************************************************************************
+// namespace extensions
+namespace Exiv2
+{
+    EXIV2API bool enableBMFF(bool enable = true);
+
+    class Iloc;  // We'll define this in bmffimage.cpp
+
+    // *****************************************************************************
+    // class definitions
+
+    // Add Base Media File Format to the supported image formats
+    namespace ImageType
+    {
+        const int bmff = 15;  //!< BMFF (bmff) image type (see class BMFF)
+    }
+
+    /*!
+      @brief Class to access BMFF images.
+     */
+    class EXIV2API BmffImage : public Image
+    {
+    public:
+        //! @name Creators
+        //@{
+        /*!
+          @brief Constructor to open a BMFF image. Since the
+              constructor can not return a result, callers should check the
+              good() method after object construction to determine success
+              or failure.
+          @param io An auto-pointer that owns a BasicIo instance used for
+              reading and writing image metadata. \b Important: The constructor
+              takes ownership of the passed in BasicIo instance through the
+              auto-pointer. Callers should not continue to use the BasicIo
+              instance after it is passed to this method.  Use the Image::io()
+              method to get a temporary reference.
+          @param create Specifies if an existing image should be read (false)
+              or if a new file should be created (true).
+         */
+        BmffImage(BasicIo::AutoPtr io, bool create);
+        //@}
+
+        //@{
+        BmffImage(BasicIo::AutoPtr io, size_t start, size_t count);
+        //@}
+
+        //@{
+        /*!
+          @brief parse embedded tiff file
+          @param root_tag root of parse tree Tag::root, Tag::cmt2 etc.
+          @param length tiff block length
+          @param start offset in file (default, io_->tell())
+         @
+         */
+        void parseTiff(uint32_t root_tag, uint32_t length);
+        void parseTiff(uint32_t root_tag, uint32_t length,uint32_t start);
+        //@}
```

Comment:
The docstring: 
> `@param start offset in file (default, io_->tell())`

Kind of suggests to me to expect that the two function are identical, except that one has a non-default `start`. 
But they actually have function bodies that differ by a bit more.   
What do you think of only having a:
```cpp
void parseTiff(uint32_t root_tag, uint32_t length,uint32_t start=0);
``` 
and making the function body work for both cases?   
Would avoid a bit of code duplication too :blush: 

Also, should we move this into the `private` part next to the `boxhandler` ?

Reply:
Yes.  I had a version of the code with a default start=0 and then I decided to leave your code in place and have a different function.  The two functions have the same result, however their plumbing is very different.

Reply:
fair enough, I don't have a strong opinion on this.